src/screens/ProfileScreen/elements/KarmaScore.tsx

```diff
@@ -1,35 +1,101 @@
 import * as React from 'react';
 import {Text, View} from 'react-native';
 import FastImage from 'react-native-fast-image';
-import {colors} from '../../../utils/colors/index';
-import BetterScore from '../../../assets/better-score.png';
+import Tooltip from 'react-native-walkthrough-tooltip';
+import {TouchableWithoutFeedback} from 'react-native-gesture-handler';
+import BetterScore from '../../../assets/Logogram.png';
+import MemoIcQuestionMark from '../../../assets/icons/Ic_question_mark';
+import {normalize} from '../../../utils/fonts';
+import {CircleGradient} from '../../../components/Karma/CircleGradient';
 
 type KarmaScoreProps = {
   score: number;
 };
 
 export const KarmaScore = ({score}: KarmaScoreProps) => {
+  const [isTooltipShown, setIsTooltipShown] = React.useState(false);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `isTooltipShown` state variable is initialized as `false`, which is good. However, it's important to ensure that this state is reset to `false` when the component unmounts to prevent potential memory leaks or unexpected behavior. You can achieve this by using a cleanup function in a `useEffect` hook.

```diff
  export const KarmaScore = ({score}: KarmaScoreProps) => {
+   React.useEffect(() => {
+     return () => {
+       setIsTooltipShown(false);
+     };
+   }, []);
    const [isTooltipShown, setIsTooltipShown] = React.useState(false);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->